torchao/quantization/observer.py

```diff
@@ -124,16 +140,25 @@ def calculate_qparams(self) -> Tuple[torch.Tensor, torch.Tensor]:
         """
         pass
 
+
 class AffineQuantizedMinMaxObserver(AffineQuantizedObserverBase):
     def forward(self, input: torch.Tensor):
         if input.numel() == 0:
             return input
 
         input_detached = input.detach()
         if self.block_size is None:
-            self.block_size = get_block_size(input_detached.shape, self.granularity_type)
+            block_size = get_block_size(input_detached.shape, self.granularity_type)
+            if not isinstance(self.granularity_type, PerTensor):
```

Comment:
this is interesting, maybe we can just always use granularity_type, and just calculate block_size in forward? will also need to have a check in forward to make sure block_size is compatible with existing min_val/max_val shape I think